build-package.sh

```diff
@@ -429,9 +430,10 @@ termux_add_package_to_built_packages_list() {
 
 # Check if the package is in the compiling list
 termux_check_package_in_building_packages_list() {
-	[ ! -f "$TERMUX_BUILD_PACKAGE_CALL_BUILDING_PACKAGES_LIST_FILE_PATH" ] && \
+	[[ ! -f "$TERMUX_BUILD_PACKAGE_CALL_BUILDING_PACKAGES_LIST_FILE_PATH" ]] && \
 		termux_error_exit "ERROR: file '$TERMUX_BUILD_PACKAGE_CALL_BUILDING_PACKAGES_LIST_FILE_PATH' not found."
-	grep -q "^${1}$" "$TERMUX_BUILD_PACKAGE_CALL_BUILDING_PACKAGES_LIST_FILE_PATH"
+	# slightly faster than `grep -q $word $file`
+	[[ $'\n'"$(<"$TERMUX_BUILD_PACKAGE_CALL_BUILDING_PACKAGES_LIST_FILE_PATH" || :)"$'\n' == *$'\n'"$1"$'\n'* ]]
```

Comment:
```suggestion
	[[ $'\n'"$(<"$TERMUX_BUILD_PACKAGE_CALL_BUILDING_PACKAGES_LIST_FILE_PATH")"$'\n' == *$'\n'"$1"$'\n'* ]]
```

While studying your new faster check algorithm, I noticed that it works differently depending on the shell you are running. I checked it on the `zsh` shell and on the `bash` shell (the default shell). On the `zsh` shell it works fine, but if you try to run it on `bash`, the algorithm always returns a negative result. My test showed that this is because the `$(<"./files" || :)` algorithm on the `bash` shell always returns empty, regardless of whether the file exists or not. There is no such problem on the `zsh` shell. Example test:
```bash
➜  ~ # zsh shell
➜  ~ cat ./building
pacman
test
glibc
➜  ~ echo "$(<"./building" || :)"
pacman
test
glibc
➜  ~ bash
~ $ # bash shell
~ $ echo "$(<"./building" || :)"

~ $
```

Reply:
It is weird because it works fine with bash on my machine in both regular environment and docker. I'll check it out.

Reply:
What bash version do you use?

Reply:
> What bash version do you use?

I ran tests in Termux on the `bash-5.2.37-2` shell and on the `bash-glibc-5.2.26-0` shell.

Reply:
It is weird, because it is worked in my case.
```
~ $ echo asdasdzxc > file
~ $ echo "$(<file)"; echo status $?
asdasdzxc
status 0
~ $ bash --version
GNU bash, version 5.2.37(1)-release (aarch64-unknown-linux-android)
...
```

Reply:
Can you please post output of this? 
```
echo "shell: $BASH $BASH_VERSION"
echo "posix: $-"
echo "POSIXLY_CORRECT=${POSIXLY_CORRECT-<unset>}"
echo "pwd = $PWD"
[[ -e ./building ]] && echo "file exists"
```

Reply:
You are completely right, fixed it. Sorry.
Somehow I missed this.

Reply:
Seems like it is the only place I used this weird syntax construction, so I do not need to fox it anywhere else in recent PRs or commits. I am glad you caught this. Thank you.